packages/flutter_tools/test/integration.shard/analyze_size_test.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:process/process.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import '../src/common.dart';
+
+const String apkDebugMessage = 'A summary of your APK analysis can be found at: ';
+const String iosDebugMessage = 'A summary of your iOS bundle analysis can be found at: ';
+
+void main() {
+  test('--analyze-size flag produces expected output on hello_world for Android', () async {
+    final String woringDirectory = globals.fs.path.join(getFlutterRoot(), 'examples', 'hello_world');
+    final String flutterBin = globals.fs.path.join(getFlutterRoot(), 'bin', 'flutter');
+    final ProcessResult result = await const LocalProcessManager().run(<String>[
+      flutterBin,
+      'build',
+      'apk',
+      '--analyze-size',
+      '--target-platform=android-arm64'
+    ], workingDirectory: globals.fs.path.join(getFlutterRoot(), 'examples', 'hello_world'));
+
+    print(result.stdout);
+    print(result.stderr);
+    expect(result.stdout.toString(), contains('app-release.apk (total compressed)'));
+
+    final String line = result.stdout.toString()
+      .split('\n')
+      .firstWhere((String line) => line.contains(apkDebugMessage));
+
+    final String outputFilePath = line.split(apkDebugMessage).last.trim();
+    expect(globals.fs.file(globals.fs.path.join(woringDirectory, outputFilePath)), exists);
+    expect(result.exitCode, 0);
+  });
+
+  test('--analyze-size flag produces expected output on hello_world for iOS', () async {
```

Comment:
do we need to add more tests for windows, macos, and linux?

Reply:
We don't yet have CI for the desktop targets :(